src/backend/src/lib.rs

```diff
@@ -528,9 +527,8 @@ pub fn add_user_credential(request: AddUserCredentialRequest) -> AddUserCredenti
                 &mut user_profile_model,
             )
```

Comment:
```suggestion
            ).into()
```
I would recommend using normal rust Results in the inner implementation.  These custom Result types exist only because of an annoying aspect of Candid bindings; they don't benefit the rest of the code.  And at some point Candid might make this easier, at which point we can throw the Result types away.  So let's make it easy to remove them.

Reply:
reverted!